portal-impl/src/com/liferay/portal/service/impl/WorkflowDefinitionLinkLocalServiceImpl.java

```diff
@@ -365,13 +380,24 @@ public WorkflowDefinitionLink updateWorkflowDefinitionLink(
 				className, classPK, typePK, workflowDefinitionName,
 				workflowDefinitionVersion);
 		}
+		else if (StringUtil.equals(
```

Comment:
Why are you deleting the entry here to add a new one? The old behavior of this method works like an UPSERT, if the entry doesn't exist it is created and updated, if it exists it is just updated.